dbms/src/Storages/StorageKafka.cpp

```diff
@@ -89,7 +91,17 @@ class ReadBufferFromKafkaConsumer : public ReadBuffer
         // Consume message and mark the topic/partition offset
         // The offsets will be committed in the insertSuffix() method after the block is completed
         // If an exception is thrown before that would occur, the client will rejoin without comitting offsets
-        BufferBase::set(reinterpret_cast<char *>(msg->payload), msg->len, 0);
+        if (format_name == "Native" || format_name == "RowBinary" || format_name == "Values" || format_name == "CapnProto")
```

Comment:
This explicit list looks ugly, especially for this branch.
We should add a property of formats "rows must be separated by newlines" and use it here.